src/coreclr/vm/fcall.h

```diff
@@ -889,6 +905,8 @@ void HCallAssert(void*& cache, void* target);
 #define FC_COMMON_PROLOG(target, assertFn) FCALL_TRANSITION_BEGIN()
 #define FC_CAN_TRIGGER_GC()
 #define FC_CAN_TRIGGER_GC_END()
+#define FC_CAN_TRIGGER_GC_HAVE_THREAD(thread)
+#define FC_CAN_TRIGGER_GC_HAVE_THREADEND(thread)
```

Comment:
```suggestion
```
These are debug-only macros. We do not need special optimized variants that take Thread*.

(Also, passing around current `Thread*` mattered a lot before current thread getter was converted to a regular thread static. It matters a lot less now.)

Reply:
> Also, passing around current `Thread*` mattered a lot before current thread getter was converted to a regular thread static. It matters a lot less now

I have measured a noticeable difference in perf when the GetThread was called multiple times in the IL_Throw.

Reply:
> These are debug-only macros. We do not need special optimized variants that take Thread*.

Ok, makes sense

Reply:
Hmm, they are actually not debug only, they are defined as empty for !ENABLE_CONTRACTS

Reply:
> I have measured a noticeable difference in perf when the GetThread was called multiple times in the IL_Throw.

`GetThread` should be just a couple of instructions and the C++ compiler should be able to CSE redundant calls. I would not expect that we are at the point where a couple more instructions in IL_Throw makes a noticeable difference.

Reply:
> Hmm, they are actually not debug only, they are defined as empty for !ENABLE_CONTRACTS

ENABLE_CONTRACTS == DEBUG